File: src/leetcode/java/MaxConsecutiveOnes_485.java
package leetcode.java;

public class MaxConsecutiveOnes_485 {
  public int findMaxConsecutiveOnes(int[] nums) {
    int max = 0, cnt = 0;
    for (int i : nums) {
      if (i == 1) cnt++;
      else {
        max = Math.max(cnt, max);
        cnt = 0;
      }
    }
    return Math.max(cnt, max);
  }
}

File: src/leetcode/python/MaxConsecutiveOnes_485.py
class MaxConsecutiveOnes_485(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mx, cnt = 0, 0
        for i in nums:
            if i == 1:
                cnt += 1
            else:
                mx = max(mx, cnt)
                cnt =0
        return max(mx, cnt)
